dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,279 @@
+        for channel in self.channel_dict.keys():
+            if self.request_status(channel)!=None:
+                self.mode_dict[channel]='streaming'
+
+                try: 
+                    if self.set_central_frequency(channel=channel, cf=800e6)==False:
+                       self.mode_dict[channel]='triggered'
+                       if self.set_central_frequency(channel=channel, cf=800e6)==False:
+                            self.mode_dict[channel]=None
+                            raise core.exceptions.DriplineGenericDAQError("Stream writer of Psyllid instance for channel {} is not configured correctly".format(channel))
+
+                    if self.get_number_of_streams(channel)!=1:
+                        self.mode_dict[channel]=None
+                        raise core.exceptions.DriplineGenericDAQError("Streams of Psyllid instance for channel {} are not configured correctly".format(channel))
+
+                except core.exceptions.DriplineGenericDAQError:
+                    logger.warning('No matching Psyllid instance for channel {} present'.format(channel))
+                except Exception as e:
+                    logger.error('Unexpected error during startup')
+                    raise 
+
+        # Summary after startup
+        logger.info('Status of channels: {}'.format(self.status_value_dict))
+        logger.info('Set central frequencies: {}'.format(self.freq_dict))
+        logger.info('Streaming or triggered mode: {}'.format(self.mode_dict))
+
+
+    def request_status(self, channel):
+        logger.info('Checking Psyllid status of channel {}'.format(channel))
+        try:
+            result = self.provider.get(self.queue_dict[channel]+'.daq-status', timeout=10)
+        except core.exceptions.DriplineError:
+            logger.warning('Psyllid instance for channel {} is not running or returned error'.format(channel))
+            self.status_dict[channel]=None
+            self.status_value_dict[channel]=None
+            return self.status_value_dict[channel]
+        except Exception as e:
+            logger.error('Something else went wrong')
+            raise
+        else:
+            self.status_dict[channel] = result['server']['status']
+            self.status_value_dict[channel] = result['server']['status-value']
+            logger.info('Psyllid is running. Status is {}'.format(self.status_dict[channel]))
+            logger.info('Status in numbers: {}'.format(self.status_value_dict[channel]))
+            return self.status_value_dict[channel]
+
+
+
+    def activate(self, channel):
+        if self.status_value_dict[channel] == 0:
+            logger.info('Activating Psyllid instance for channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel], 'activate-daq')
+        else:
+            logger.warning('Cannot activate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
```

Comment:
Psyllid used to have problems with changing the daq-status and not getting stuck somewhere. Then activating did not result in an error and still the status could be wrong afterwards.
Currently this is not the case anymore but I still don't feel totally comfortable with just assuming the correct status after requesting a change as long as psyllid is still under development.
If activating psyllid fails and results in an error the error message from psyllid is forwarded to the operator (will double check). So I don't think we need an exception here.